src/order/filter/basic.lean

```diff
@@ -1758,6 +1764,51 @@ lemma ne_bot.comap_of_range_mem {f : filter β} {m : α → β}
   (hf : ne_bot f) (hm : range m ∈ f) : ne_bot (comap m f) :=
 comap_ne_bot_iff_frequently.2 $ eventually.frequently hm
 
+@[simp] lemma comap_fst_ne_bot_iff {f : filter α} :
+  (f.comap (prod.fst : α × β → α)).ne_bot ↔ f.ne_bot ∧ nonempty β :=
+begin
+  casesI is_empty_or_nonempty β,
+  { rw [filter_eq_bot_of_is_empty (f.comap _), ← not_iff_not]; [simp *, apply_instance] },
+  { simp [comap_ne_bot_iff_frequently, h] }
+end
+
+@[instance] lemma comap_fst_ne_bot [nonempty β] {f : filter α} [ne_bot f] :
```

Comment:
Is there a reason why you are using `@[instance] lemma` instead of just `instance`, here and in most lemmas below? (but not in `Coprod_ne_bot`, surprisingly).

Reply:
With `@[instance] lemma`, the lemma is not `protected`. For `Coprod_ne_bot` this is a typo.